scico/linop/radon_svmbir.py

```diff
@@ -63,6 +70,11 @@ def __init__(
                 f"Only 2D and 3D inputs are supported, but input_shape was {input_shape}"
             )
 
+        if is_masked:
+            self.roi_radius = None
```

Comment:
Tests need to cover both possible values of `is_masked`.

Reply:
I added tests that cover both values of `is_masked`. (except `test_prox_cg' since we don't expect it to pass if there is masking)

Reply:
There's no way of making a working `test_prox_cg` with masking enabled?

Reply:
One way (although a little hacky) could be to find the mask either by back-projecting a sinogram of 1's or by thresholding a reconstruction and then use the mask for comparing the norms of the two proxes. 

Let me know whether this sounds reasonable.

Reply:
It's perhaps a bit hacky for the main package code, but that would be fine for a test.

Reply:
I expanded the test to include the masked case. For the masked case, the comparison between cg and inbuilt-prox is done in the masked region. The mask is computed by back-projecting 1's.

However, I noticed that for the small input case, the mask is the full image. Should we have a larger test case?

Another thing I noticed was that svmbir.prox returns ndarray whereas cg returns jax-device array. Is this by design, or shall we make them both the same data type?

Reply:
> However, I noticed that for the small input case, the mask is the full image. Should we have a larger test case?

I think it's fine as-is.

> Another thing I noticed was that svmbir.prox returns ndarray whereas cg returns jax-device array. Is this by design, or shall we make them both the same data type?

I would expect consistency in data type, but perhaps there's a good reason for this choice. @Michael-T-McCann and @tbalke, can you comment on this?

Reply:
@smajee We recently implemented the `ensure_on_device` routine when we realized that we need to restrict the _input_ types somewhat. (Making it work with ndarrays became impractical). Our conclusion was that we build in some safeguards but in the end the user has to have some foresight since unnecessary conversion may lead to performance slowdowns. 

I think the current stage on what type our routines _return_ is: they may return ndarrays, device arrays, or block arrays but we don't seem to be very precise and consistent about it. I think most routines will not return nd-arrays though (maybe svmbir stuff is the only exception) because once on device, always on device.

So that makes me lean towards casting the svmbir output to device array. Especially, when considering that the prox may return an ndarray somewhere deep inside the admm iterations, it seem to break our assumption that the optimizers are internally only working with device arrays.

Reply:
At some point we should discuss whether it's possible to be more consistent on return types.

The `svmbir` and `bm3d` interfaces are special cases in that they can't be run a GPU, but I agree with your concern about prox returning an ndarray in a context in which device array is expected. My immediate response would be that making the output type the same as the input is a good approach, but this deserves a bit more consideration. (And we should make sure that the same approach is followed for the `bm3d` interface.)

Reply:
I don't think there is a special reason to return an `ndarray`. Agreed that matching the input type makes sense.